react-components/src/data/search/queryBuilder/Group.js

```diff
@@ -0,0 +1,45 @@
+import build from "../../../util/DebugIDUtil";
+import AddBtn from "./AddBtn";
+import Condition from "./Condition";
+import DeleteBtn from "./DeleteBtn";
+import ids from "../ids";
+
+import React, { Fragment } from 'react';
+
+/**
+ * A component which allows users to create different groupings in QueryBuilder
+ * such as Any and All
+ */
+function Group(props) {
+    let {
+        root,
+        fields,
+        onRemove,
+        helperProps,
+        helperProps: {
+            parentId,
+            classes
+        }
+    } = props;
+
+    let baseId = build(parentId, fields.name);
+
+    return (
+        <Fragment>
+            <AddBtn onClick={() => fields.push({})}
+                    id={build(baseId, ids.addConditionBtn)}/>
+            {!root && <DeleteBtn onClick={onRemove}
+                                 id={build(parentId, ids.deleteConditionBtn)}/>}
```

Comment:
Minor suggestion: since the rows can have variable lengths, would the add/delete buttons line-up better if they were at the start of each row instead of at the end? 🤔

Reply:
If they were in the front, would we want them to still line up with each group? (It'd take a bit more effort to get the Add/Delete buttons by groups in front, so use your imagination for now 😉 )
![image](https://user-images.githubusercontent.com/8909156/44160597-ba2b1600-a06f-11e8-8c0c-15534077e991.png)


I could also line up the Delete buttons at the end.  I thought it might be too easy for users to unintentionally click the Delete button on a group though and lose a bit of work, so maybe leave those where they are? 
![image](https://user-images.githubusercontent.com/8909156/44125219-3532321e-9fe6-11e8-8fd1-95ad99c8b5d8.png)

Reply:
I think I mentioned earlier in my comment, but is it possible add more padding between the `add` and `delete` button. They are almost touching each other...

Reply:
How does the padding look here?
![image](https://user-images.githubusercontent.com/8909156/44179600-9e923080-a0ac-11e8-89f7-0d4e78aa5979.png)